File: guests/go/panels/panels.go
// The panels conformance scene, Go port — the auxiliary-window
// grammar via the chain spelling. See guests/rust/panels.rs and
// tools/scenes/panels.steps.
package panels

import (
	kaya "dev.kaya/bindings/go"
)

func App() *kaya.App {
	app := kaya.NewApp()

	var status kaya.Signal[string]
	app.Build(func(tx *kaya.Tx) {
		tx.Window(0).Title("panels")
		status = tx.Signal("two panels")

		tx.Mount(tx.Column(func() {
			tx.Label(status) // label#0
		}))

		inspector := tx.CreateWindow(1).
			Title("inspector").
			Size(480, 320).
			VetoClose(true).
			OnCloseRequested(func(tx *kaya.Tx) {
				tx.Write(status, "close requested")
				tx.DestroyWindow(1)
			})
		aux := tx.Column(func() {
			caption := tx.Signal("inspector pane")
			tx.Label(caption) // label#1
		})
		tx.MountIn(inspector.Id(), aux)
	})

	return app
}

File: guests/java/dev/kaya/guests/Panels.java
package dev.kaya.guests;

import dev.kaya.KayaApp;

/**
 * The panels scene from the JVM — guests/rust/panels.rs,
 * tools/scenes/panels.steps.
 */
public final class Panels {
    public static void app() {
        KayaApp app = new KayaApp();

        KayaApp.Signal<String> status = app.build(tx -> {
            tx.window(0).title("panels");
            KayaApp.Signal<String> s = tx.signal("two panels");
            tx.mount(tx.column(() -> {
                tx.label(s); // label#0
            }));

            tx.createWindow(1)
                    .title("inspector")
                    .size(480.0, 320.0)
                    .vetoClose(true)
                    .onCloseRequested(tx2 -> {
                        tx2.write(s, "close requested");
                        tx2.destroyWindow(1);
                    });
            tx.mountIn(1, tx.column(() -> {
                KayaApp.Signal<String> caption = tx.signal("inspector pane");
                tx.label(caption); // label#1
            }));
            return s;
        });

        if (status == null) throw new IllegalStateException();

        app.dispatchLoop();
    }

    private Panels() {}
}
